schemas/config/3.0.schema.json

```diff
@@ -1,2600 +1,2869 @@
-                            "type": "string",
-                            "default": null
-                        },
-                        "user_agent":{
-                            "description": "UserAgent is the user agent to for the client to use when connecting to Kafka, overriding the default franz-go/<runtime.Version()>/<hostname>. Setting a UserAgent allows authorizing based on the aws:UserAgent condition key; see the following link for more details: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-useragent",
-                            "type": "string",
-                            "default": null
-                        }
-                    }
-                }
-            }
-        },
-        "BoltDB_J": {
-            "description": "Boltdb jobs driver",
-            "type": "object",
-            "properties": {
-                "permissions": {
-                    "type": "integer",
-                    "default": "0777"
-                }
-            }
-        }
-    }
+		},
+		"HostAndPortWithTCP": {
+			"description": "Host and port with tcp:// prefix",
+			"type": "string",
+			"pattern": "^(((tcp://[0-9a-zA-Z_.-]+|)|\\$\\{([^}]+)\\}):([0-9]{1,5}|\\$\\{([^}]+)\\}))|\\$\\{([^}]+)\\}$",
+			"examples": [
+				"tcp://127.0.0.1:443",
+				"${TCP:-tcp://127.0.0.1:443}",
+				"tcp://127.0.0.1:${TCP_PORT}"
+			]
+		},
+		"HostAndPort": {
+			"description": "Host and port",
+			"type": "string",
+			"pattern": "^((([0-9a-zA-Z_.-]+|)|\\$\\{([^}]+)\\}):([0-9]{1,5})|\\$\\{([^}]+)\\})|\\$\\{([^}]+)\\}$",
+			"examples": [
+				"127.0.0.1:443",
+				":8080",
+				"0.0.0.0:${HTTP_PORT:-8080}",
+				"${HTTP_HOST:-127.0.0.1:8000}"
+			]
+		},
+		"LogMode": {
+			"description": "Logging mode",
+			"type": "string",
+			"enum": [
+				"development",
+				"production",
+				"raw"
+			]
+		},
+		"LogLevel": {
+			"description": "Logging level",
+			"type": "string",
+			"enum": [
+				"debug",
+				"info",
+				"warn",
+				"error",
+				"panic"
+			]
+		},
+		"LogEncoding": {
+			"description": "Encoding format",
+			"type": "string",
+			"enum": [
+				"console",
+				"json"
+			]
+		},
+		"LogLineEnding": {
+			"description": "Line-ending to use for logging.",
+			"type": "string",
+			"default": "\n"
+		},
+		"LogOutput": {
+			"type": "string",
+			"examples": [
+				"stdout",
+				"stderr",
+				"/var/log/rr_errors.log"
+			]
+		},
+		"HashmapString": {
+			"description": "A hash map that only allows string values.",
+			"type": "object",
+			"patternProperties": {
+				"^[a-zA-Z0-9._-]+$": {
+					"type": "string"
+				}
+			},
+			"additionalProperties": false
+		},
+		"HashmapInteger": {
+			"description": "A hash map that only allows integer values.",
+			"type": "object",
+			"patternProperties": {
+				"^[a-zA-Z0-9._-]+$": {
+					"type": "integer"
+				}
+			}
+		},
+		"DatabaseProperties": {
```

Comment:
Some of the Bolt properties were being reused, but not the entire object, so I created a definition just for the reused props.